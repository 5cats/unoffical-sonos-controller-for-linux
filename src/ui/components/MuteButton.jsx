	
import React from 'react/addons';
import { Cursor, ImmutableOptimizations }  from 'react-cursor';
import EventableMixin from '../mixins/EventableMixin';

class MuteButton {

	render () {

		var id = this.props.id || '';
		var src = this.props.muted.value ? 'svg/mute_on.svg' : 'svg/mute_off.svg'; 

		return (
			<img 
			onClick={this._toggleMute}
			className="mute-button"
			id={id}
			src={src} />
		);
	}

	_toggleMute () {
		this.trigger('volume:togglemute', this.props.id);
	}
}

MuteButton.prototype.displayName = "MuteButton";
MuteButton.prototype.mixins = [
	ImmutableOptimizations(['cursor']),
	EventableMixin
];
MuteButton.prototype.propTypes = {
	muted: React.PropTypes.instanceOf(Cursor).isRequired
};
export default React.createClass(MuteButton.prototype);