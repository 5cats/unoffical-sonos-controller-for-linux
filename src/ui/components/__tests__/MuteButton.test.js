import React from 'react';
import { MuteButton } from '../MuteButton';
import { render, mount } from 'enzyme';

describe('MuteButton', () => {
    it('renders on', () => {
        const props = {
            muted: false,
        };
        const context = render(<MuteButton {...props} />);
        expect(context).toMatchSnapshot();
    });

    it('renders off', () => {
        const props = {
            muted: true,
        };
        const context = render(<MuteButton {...props} />);
        expect(context).toMatchSnapshot();
    });

    it('clickhandler', () => {
        const props = {
            clickHandler: jest.fn(),
        };
        const context = mount(<MuteButton {...props} />);
        context.find('.mute-button').simulate('click');
        expect(props.clickHandler).toHaveBeenCalled();

        context.unmount();
    });
});
