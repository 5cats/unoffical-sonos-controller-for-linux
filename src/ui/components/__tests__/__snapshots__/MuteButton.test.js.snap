// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MuteButton renders off 1`] = `
<img
  class="mute-button"
  src="svg/mute_on.svg"
/>
`;

exports[`MuteButton renders on 1`] = `
<img
  class="mute-button"
  src="svg/mute_off.svg"
/>
`;
